### PR TITLE
tiproxy: add description on graceful-wait and signing-cert

### PR DESCRIPTION
**Pull request by xhebox**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.5 (TiDB Operator 1.5 versions)
- [ ] v1.4 (TiDB Operator 1.4 versions)
- [ ] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: zh/deploy-tiproxy.md [hunk 1]
```diff
@@ -39,13 +39,33 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
+          [log]
+          level = "info"
+          [proxy]
+          graceful-wait-before-shutdown = 120
     ```
 
-    要获取更多可配置的 TiProxy 配置参数，请参考 [TiProxy 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-configuration)。
+    推荐把 `graceful-wait-before-shutdown` 设置为 120 秒。 要获取更多可配置的 TiProxy 配置参数，请参考 [TiProxy 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-configuration)。
+
+4. 如果没有开启集群 TLS ，还需要手动配置 TiDB　的 `signing-cert` 参数：
```
Comment (ran-huang) [inline zh/deploy-tiproxy.md:51]:
```suggestion
4. 如果没有开启集群 TLS，还需要手动配置 TiDB 的 `signing-cert` 参数：
```

File: zh/deploy-tiproxy.md [hunk 2]
```diff
@@ -39,13 +39,33 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
-            level = "info"
+        config: |
+          [log]
+          level = "info"
+          [proxy]
+          graceful-wait-before-shutdown = 120
     ```
 
-    要获取更多可配置的 TiProxy 配置参数，请参考 [TiProxy 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-configuration)。
+    推荐把 `graceful-wait-before-shutdown` 设置为 120 秒。 要获取更多可配置的 TiProxy 配置参数，请参考 [TiProxy 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-configuration)。
```
Comment (djshow832) [inline zh/deploy-tiproxy.md:49]: 如果不是云上，没必要设置。

File: zh/deploy-tiproxy.md [hunk 3]
```diff
@@ -39,13 +39,33 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
+              secretName: sessioncert
+        additionalVolumeMounts:
+          - name: sessioncert
+            mountPath: /var/session
+        config: |
+          session-token-signing-cert = "/var/session/tls.crt"
+          session-token-signing-key = "/var/session/tls.key"
+    ```
+
+    详见 [TiDB 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tidb-configuration-file#session-token-signing-cert-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5)
```
Comment (ran-huang) [inline zh/deploy-tiproxy.md:68]:
```suggestion
    详见 TiDB 配置文档 [`session-token-signing-key`](https://docs.pingcap.com/zh/tidb/v7.6/tidb-configuration-file#session-token-signing-cert-从-v640-版本开始引入)。
```

File: zh/deploy-tiproxy.md (hunk 1 above)
Comment (djshow832) [inline zh/deploy-tiproxy.md:51]:
```suggestion
4. 如果没有开启集群 TLS ，还需要手动配置 TiDB 的 [`session-token-signing-cert`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-cert-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) 和 [`session-token-signing-key`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-key-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5):
```

File: zh/deploy-tiproxy.md (hunk 2 above)
Comment (djshow832) [reply zh/deploy-tiproxy.md:49]: 要设置的是 TiDB 的 graceful-wait-before-shutdown

Comment (djshow832) [reply zh/deploy-tiproxy.md:49]: 这里的场景是用户自己用 operator 部署。tidbcloud 的不用写，不是用户部署的。

File: zh/deploy-tiproxy.md [hunk 4]
```diff
@@ -39,13 +39,33 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
+          [log]
+          level = "info"
+          [proxy]
+          graceful-wait-before-shutdown = 120
     ```
 
-    要获取更多可配置的 TiProxy 配置参数，请参考 [TiProxy 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-configuration)。
+    推荐把 `graceful-wait-before-shutdown` 设置为 120 秒。 要获取更多可配置的 TiProxy 配置参数，请参考 [TiProxy 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-configuration)。
+
+4. 如果没有开启集群 TLS ，还需要手动配置 TiDB 的 [`session-token-signing-cert`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-cert-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) 和 [`session-token-signing-key`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-key-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5):
```
Comment (djshow832) [inline zh/deploy-tiproxy.md:51]:
```suggestion
4. 如果开启了[集群 TLS](enable-tls-between-components.md)，则跳过这一步；如果没有开启集群 TLS，还需要生成自签名证书，并手动配置 TiDB 的 [`session-token-signing-cert`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-cert-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) 和 [`session-token-signing-key`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-key-%E4%BB%8E-v640-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5):
```

**Review by djshow832 — COMMENTED** [review]
configure-a-tidb-cluster.md 里也要提示用户做这些操作，可以用引用的方式

File: zh/configure-a-tidb-cluster.md [hunk 5]
```diff
@@ -262,6 +262,8 @@ spec:
     config:
 ```
 
+如果未开启集群 TLS，还需要配置额外参数。详细的配置步骤见 [为已有 TiDB 集群部署负载均衡 TiProxy](https://docs.pingcap.com/tidb-in-kubernetes/stable/deploy-tiproxy)。
```
Comment (djshow832) [inline zh/configure-a-tidb-cluster.md:265]: 不管开没开启 TLS，TiDB 的 graceful-wait-before-shutdown 都是要配的。
去掉“如果未开启集群 TLS”。

Comment (djshow832) [reply zh/configure-a-tidb-cluster.md:265]:
```suggestion
部署 TiProxy 时，还需要给 TiDB 配置额外参数，详细的配置步骤见 [为已有 TiDB 集群部署负载均衡 TiProxy](deploy-tiproxy.md)。
```

**Review by djshow832 — APPROVED** [review]

File: zh/configure-a-tidb-cluster.md [hunk 6]
```diff
@@ -262,6 +262,8 @@ spec:
     config:
 ```
 
+部署 TiProxy 时，还需要给 TiDB 配置额外参数，详细的配置步骤见 [为已有 TiDB 集群部署负载均衡 TiProxy](deploy-tiproxy.md)。
```
Comment (ran-huang) [inline zh/configure-a-tidb-cluster.md:265]:
```suggestion
部署 TiProxy 时，还需要给 TiDB 配置额外参数，详细的配置步骤见[为已有 TiDB 集群部署负载均衡 TiProxy](deploy-tiproxy.md)。
```

File: zh/deploy-tiproxy.md [hunk 7]
```diff
@@ -39,14 +39,43 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
+        config: |
+          [log]
+          level = "info"
     ```
 
     要获取更多可配置的 TiProxy 配置参数，请参考 [TiProxy 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-configuration)。
 
+4. 配置 TidbCluster CR 中 `spec.tidb` 的相关参数：
+
+    + 推荐设置 TiDB 的 `graceful-wait-before-shutdown` 大于应用程序最长的事务的持续时间，配合 TiProxy 的链接迁移，详见 [TiProxy 使用限制](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-overview#使用限制)。
```
Comment (ran-huang) [inline zh/deploy-tiproxy.md:51]:
```suggestion
    + 推荐设置 TiDB 的 `graceful-wait-before-shutdown` 大于应用程序最长的事务的持续时间，配合 TiProxy 的连接迁移，详见 [TiProxy 使用限制](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-overview#使用限制)。
```

File: zh/deploy-tiproxy.md [hunk 8]
```diff
@@ -39,14 +39,43 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
+
+       ```yaml
+       spec:
+         tidb:
+           config: |
+             graceful-wait-before-shutdown = 30
+       ```
+
+    + 如果开启了[集群 TLS](enable-tls-between-components.md)，则跳过这一步；如果没有开启集群 TLS，还需要生成自签名证书，并手动配置 TiDB 的 [`session-token-signing-cert`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-cert-从-v640-版本开始引入) 和 [`session-token-signing-key`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-key-从-v640-版本开始引入):
```
Comment (ran-huang) [inline zh/deploy-tiproxy.md:60]:
```suggestion
    + 如果开启了[集群 TLS](enable-tls-between-components.md)，则跳过这一步；如果没有开启集群 TLS，还需要生成自签名证书，并手动配置 TiDB 的 [`session-token-signing-cert`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-cert-从-v640-版本开始引入) 和 [`session-token-signing-key`](https://docs.pingcap.com/zh/tidb/stable/tidb-configuration-file#session-token-signing-key-从-v640-版本开始引入)：

File: zh/deploy-tiproxy.md [hunk 9]
```diff
@@ -39,14 +39,43 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
+             - name: sessioncert
+               secret:
+                 secretName: sessioncert-secret
+           additionalVolumeMounts:
+             - name: sessioncert
+               mountPath: /var/session
+           config: |
+             session-token-signing-cert = "/var/session/tls.crt"
+             session-token-signing-key = "/var/session/tls.key"
+       ```
```
Comment (ran-huang) [inline zh/deploy-tiproxy.md:75]:
```suggestion
        ```yaml
        spec:
          tidb:
            additionalVolumes:
              - name: sessioncert
                secret:
                  secretName: sessioncert-secret
            additionalVolumeMounts:
              - name: sessioncert
                mountPath: /var/session
            config: |
              session-token-signing-cert = "/var/session/tls.crt"
              session-token-signing-key = "/var/session/tls.key"
        ```
```

File: en/deploy-tiproxy.md [hunk 10]
```diff
@@ -39,14 +39,43 @@ If you need to deploy TiProxy for an existing TiDB cluster, follow these steps:
...
+        config: |
+          [log]
+          level = "info"
     ```
 
     For more information about TiProxy configuration, see [TiProxy Configuration](https://docs.pingcap.com/tidb/v7.6/tiproxy-configuration).
 
+4. Configure the related parameters in `spec.tidb` of the TidbCluster CR. For example:
+
+    + It is recommended to configure `graceful-wait-before-shutdown` to a value longer than the maximum duration of the transactions in your application. This is used together with TiProxy's connection migration feature. For more information, see [TiProxy Limitations](https://docs.pingcap.com/tidb/v7.6/tiproxy-overview#limitations).
```
Comment (hfxsd) [inline en/deploy-tiproxy.md:51]:
```suggestion
    + It is recommended to configure `graceful-wait-before-shutdown` to a value greater than the maximum duration of the transactions in your application. This is used together with TiProxy's connection migration feature. For more information, see [TiProxy Limitations](https://docs.pingcap.com/tidb/v7.6/tiproxy-overview#limitations).
```

File: en/deploy-tiproxy.md [hunk 11]
```diff
@@ -39,14 +39,43 @@ If you need to deploy TiProxy for an existing TiDB cluster, follow these steps:
...
+                  secretName: sessioncert-secret
+            additionalVolumeMounts:
+              - name: sessioncert
+                mountPath: /var/session
+            config: |
+              session-token-signing-cert = "/var/session/tls.crt"
+              session-token-signing-key = "/var/session/tls.key"
+        ```
+
+        For more information, refer to TiDB configuration file [`session-token-signing-cert`](https://docs.pingcap.com/tidb/stable/tidb-configuration-file#session-token-signing-cert-new-in-v640).
```
Comment (hfxsd) [inline en/deploy-tiproxy.md:77]:
```suggestion
        For more information, see [`session-token-signing-cert`](https://docs.pingcap.com/tidb/stable/tidb-configuration-file#session-token-signing-cert-new-in-v640).
```

File: zh/deploy-tiproxy.md [hunk 12]
```diff
@@ -39,14 +39,43 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
+        config: |
+          [log]
+          level = "info"
     ```
 
     要获取更多可配置的 TiProxy 配置参数，请参考 [TiProxy 配置文档](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-configuration)。
 
+4. 配置 TidbCluster CR 中 `spec.tidb` 的相关参数：
+
+    + 推荐设置 TiDB 的 `graceful-wait-before-shutdown` 大于应用程序最长的事务的持续时间，配合 TiProxy 的连接迁移，详见 [TiProxy 使用限制](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-overview#使用限制)。
```
Comment (hfxsd) [inline zh/deploy-tiproxy.md:51]:
```suggestion
    + 推荐设置 TiDB `graceful-wait-before-shutdown` 的值大于应用程序中事务的最长的持续时间，配合 TiProxy 的连接迁移。详见 [TiProxy 使用限制](https://docs.pingcap.com/zh/tidb/v7.6/tiproxy-overview#使用限制)。
```

File: zh/deploy-tiproxy.md [hunk 13]
```diff
@@ -39,14 +39,43 @@ summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署负载均衡 T
...
+                  secretName: sessioncert-secret
+            additionalVolumeMounts:
+              - name: sessioncert
+                mountPath: /var/session
+            config: |
+              session-token-signing-cert = "/var/session/tls.crt"
+              session-token-signing-key = "/var/session/tls.key"
+        ```
+
+       详见 TiDB 配置文档 [`session-token-signing-key`](https://docs.pingcap.com/zh/tidb/v7.6/tidb-configuration-file#session-token-signing-cert-从-v640-版本开始引入)。
```
Comment (hfxsd) [inline zh/deploy-tiproxy.md:77]:
```suggestion
       详见 [`session-token-signing-key`](https://docs.pingcap.com/zh/tidb/v7.6/tidb-configuration-file#session-token-signing-cert-从-v640-版本开始引入)。
```

**Review by hfxsd — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**